lib/src/code_generator/func.dart

```diff
@@ -119,21 +118,53 @@ class Func extends LookUpBinding {
         ? _exposedDartFunctionTypealias!.name
         : functionType.getDartType(w, writeArgumentNames: false);
 
-    if (exposeSymbolAddress) {
-      // Add to SymbolAddress in writer.
-      w.symbolAddressWriter.addSymbol(
-        type:
-            '${w.ffiLibraryPrefix}.Pointer<${w.ffiLibraryPrefix}.NativeFunction<$cType>>',
-        name: name,
-        ptrName: funcPointerName,
-      );
+    if (ffiNativeConfig.enabled) {
+      final assetString = ffiNativeConfig.asset != null
+          ? ", asset: '${ffiNativeConfig.asset}'"
+          : '';
+      final isLeafString = isLeaf ? ', isLeaf: true' : '';
+      s.write(
+          "@${w.ffiLibraryPrefix}.FfiNative<$cType>('$originalName'$assetString$isLeafString)\n");
+
+      s.write(
+          'external ${functionType.returnType.getDartType(w)} $enclosingFuncName(\n');
+      for (final p in functionType.parameters) {
+        s.write('  ${p.type.getDartType(w)} ${p.name},\n');
+      }
+      s.write(');\n\n');
+    } else {
+      // Write enclosing function.
+      s.write(
+          '${functionType.returnType.getDartType(w)} $enclosingFuncName(\n');
+      for (final p in functionType.parameters) {
+        s.write('  ${p.type.getDartType(w)} ${p.name},\n');
+      }
+      s.write(') {\n');
+      s.write('return $funcVarName');
+
+      s.write('(\n');
+      for (final p in functionType.parameters) {
+        s.write('    ${p.name},\n');
+      }
+      s.write('  );\n');
+      s.write('}\n');
+
+      if (exposeSymbolAddress) {
```

Comment:
Since exposeSymbolAddress is ignored incase of ffi native functions, maybe we should raise a warning log to users with both of these keys in their config.